pkgs/stdenv/cross/default.nix

```diff
@@ -53,12 +53,15 @@ in lib.init bootStages ++ [
            else buildPackages.gcc;
 
       extraNativeBuildInputs = old.extraNativeBuildInputs
+        ++ lib.optionals
+             (hostPlatform.isLinux && !buildPlatform.isLinux)
+             [ buildPackages.patchelf buildPackages.paxctl ]
+        ++ lib.optional
+             (let f = p: !p.isx86 || p.libc == "musl"; in f hostPlatform && !(f buildPlatform))
```

Comment:
This seems a little bit too much. I think I would prefer the more verbose: ```(!hostPlatform.isx86 || hostPlatform.libc == "musl" && !(!buildPlatform.isx86 || buildPlatform.libc == "musl")```. Would it hurt to just do this everywhere? It's just a hook so maybe it's smart enough to avoid the conditional completely?

Reply:
I do want to do this everywhere, but unfortunately I don't see a good way to deduplicate it between stdenv/linux and stdenv/cross since `stdenv/generic` doesn't get passed whole package sets.

Or...we could just use one of the builtin fetchers to get the new scripts so we don't even need most of nixpkgs :smiling_imp: